ui/shared/components/panel/variants/VariantIndividuals.jsx

```diff
@@ -336,46 +397,93 @@ const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById })
   const quality = Number.isInteger(genotype.gq) ? genotype.gq : genotype.qs
   const filters = genotype.filters?.join(', ') || variant.genotypeFilters
 
+  return {
+    genotype,
+    isHemiX,
+    warning,
+    previousCall,
+    hasConflictingNumAlt,
+    details,
+    showSecondaryQuality,
+    secondaryQuality,
+    quality,
+    filters,
+  }
+}
+
+const Genotype = React.memo(({ variant, individual, isCompoundHet, genesById }) => {
+  if (!variant.genotypes) {
+    return null
+  }
+
+  let genotypes = variant.genotypes[individual.individualGuid]
+  if (!genotypes) {
+    return null
+  }
+  genotypes = Array.isArray(genotypes) ? genotypes : [genotypes]
+
+  const sampleTypeOrder = [SAMPLE_TYPE_GENOME, SAMPLE_TYPE_EXOME]
+  genotypes = genotypes.sort(
+    (a, b) => sampleTypeOrder.indexOf(a.sampleType) - sampleTypeOrder.indexOf(b.sampleType),
+  ).map(
+    genotype => getAllGenotypeDetails(genotype, variant, individual, isCompoundHet, genesById),
+  )
+  const allDetails = genotypes.map(info => info.details).reduce((acc, details, index) => {
+    if (index > 0) {
+      acc.push(<Divider />)
+    }
+    return acc.concat(details)
+  }, [])
+
+  const firstGenotype = genotypes[0]
+  const legacyOtherSampleGenotypes = useMemo(() => firstGenotype.genotype?.otherSample && [
+    { genotype: firstGenotype.genotype.otherSample, isHemiX: firstGenotype.isHemiX },
+  ], [firstGenotype])
```

Comment:
`useMemo` is an antipattern here - rather than creating a new dictionary as a prop it seems like it would be much cleaner to just pass in `firstGenotype.genotype` as the prop and add conditional logic in the component to use `otherSample` instead if it is present. Or add a `useOtherSampleType` prop

Reply:
yeah that's way clearer, thank you